lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -1516,9 +1535,33 @@ def _deploy(
                     f"Exception: {error_message}"
                 )
             else:
+                if get_matched_str(error_message, NOT_SUPPORTED_VHD_PATTERN):
+                    error_message = self._replace_blob_name_as_original_vhd(
+                        environment, error_message
+                    )
+
                 plugin_manager.hook.azure_deploy_failed(error_message=error_message)
                 raise LisaException(error_message)
 
+    def _replace_blob_name_as_original_vhd(
+        self, environment: Environment, error_message: str
+    ) -> str:
+        information = self._get_environment_information(environment)
```

Comment:
Because the original vhd path was saved already, it doesn't need to recover it from information. Just use the secret function to replace it.

Reply:
I didn't quite understand. You mean the original vhd path is saved in the node_context after I modified code according to your first comment, right? If so, in the _deploy function, I still need to get it from Environment. 

For the comment "Just use the secret function to replace it.", the original vhd path is saved in the secret list already, it will be masked when it is printed. But in the error message, it is still the original format that has the SAS token.  If I understand right, you mean that I can use function replace in secret.py right?

Reply:
I added some changes.